src/game/shared/particle_property.cpp

```diff
@@ -200,6 +200,43 @@ CNewParticleEffect *CParticleProperty::Create( const char *pszParticleName, Part
 	return Create( pDef, iAttachType, iAttachmentPoint, vecOriginOffset, matOffset );
 }
 
+CNewParticleEffect* CParticleProperty::Colored_Create(const char* pszParticleName, ParticleAttachment_t iAttachType, int iAttachmentPoint, Vector colorMul, Vector vecOriginOffset, matrix3x4_t* matOffset)
```

Comment:
not sure about this. particle effects are supposed to use control points to transmit information, not edit the particle effect definition at runtime.

Reply:
The definition class data is inside the particle..lib file so I can't access it.

My work around was to create template lasers in the pcf file, one for each marine to use.

The definition uses a bunch of pointers so to create a new definition would require emulating the functions inside particle.lib.

A better solution would be to create a new particle definitions class as a clone, but all my attempts failed with conflicts in the particle.lib file.

The current process detects changes to the netprop on the client side, checks for a currently unused particle effect template index, and alters the existing particle definition template data.

Without source to particle.cpp, I'm kindof stuck unless you know of a better way to generate a definition clone.

The index is never networked, it is just a local variable to keep track of what is currently in use on the client's end.

Reply:
Can we just add the color for the laser pointer as an input control point to the particle system and then set that? That's how the game handles, for example, icons on the floor for stuff like "weld this door"

Reply:
Maybe?

Do you have an example of how and where that is done at so I can look into it?

Reply:
You mean like this?

m_pLaserPointerEffect->SetControlPoint(20, Vector(255, 255, 255));

Reply:
Doesn't work.

What indexes control "m_ColorMin" and "m_ColorMax"?

Reply:
![image](https://user-images.githubusercontent.com/4257305/197293890-bcff2708-210d-4656-a3a4-dec1cf6ebeba.png)

Open the game in tools mode via the SDK
weapon_laser_sight is in asw_weapon_fx.pcf (you'll need to extract it from the VPK)
You can set the color range to white and add a control point for the tint

Reply:
tint doesn't look like it's going to work.

A tint is a color blend, unless if there is a way to get tint to replace the color completely.

Reply:
255 0 0 shows as pink.

Reply:
and.. I broke it.

Reply:
I'm not sure if you can even access the
	Vector m_flNormColorMin;
	Vector m_flNormColorMax;
	
variables with the pcf files.

Reply:
Yeah, I don't think control points are going to work.

The m_flNormColorMax and m_flNormColorMin variables don't appear to be included in the pcf data values.

Modifying the definition may be the only way we can product these color overrides.

If we had access to the particles files then we could set up something better but I'm not seeing any other solutions. Maybe you can find something better?

Reply:
I may have found something. Tint won't work, but Remap Control Point to Vector might work.

I'm trying to program it in and see if it has desired result.

Reply:
I can get the desired results in particle editor, but the setcontrolpoint method doesn't update the color while in actual game.

Not sure what I'm doing wrong.

Reply:
ah, I get it now.

The Remap vector range was 0.0 to 1.0. I was treating it as a 0 - 255.

It is updating the color now. The result is not the same as m_flNormColorMax in definition but it is better than tint.

Reply:
Ok, I uploaded the new test code.